### PR TITLE
Documentation: Improve grammar and fix typos

### PR DESCRIPTION
**Pull request by murshed**
## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: docs/contributors/git-workflow.md [hunk 1]
```diff
@@ -41,9 +41,9 @@ git push --force-with-lease origin your-branch-name
 
 ## Keeping Your Fork Up To Date
 
-Working on pull request starts with forking the Gutenberg repository, your separate working copy. Which can easily go out of sync as new pull requests are merged into the main repository. Here your working repository is a `fork` and the main Gutenberg repository is `upstream`. When working on new pull request you should always update your fork before you do `git checkout -b my-new-branch` to work on a feature or fix.
+Working on a pull request starts with forking the Gutenberg repository, your separate working copy. Which can easily go out of sync as new pull requests are merged into the main repository. Here your working repository is a `fork` and the main Gutenberg repository is `upstream`. When working on a new pull request you should always update your fork before you do `git checkout -b my-new-branch` to work on a feature or fix.
 
-You will need to add an `upstream` remote in order to keep your fork updated.
+You will need to add an `upstream` remote in the order to keep your fork updated.
```
Comment (swissspidy) [inline docs/contributors/git-workflow.md]: This change is incorrect.

See https://dictionary.cambridge.org/grammar/british-grammar/in-order-to

File: docs/contributors/outreach.md [hunk 2]
```diff
@@ -35,7 +35,7 @@ A short list of useful articles around defining, extending, and contributing to
 
 ## Talks
 
-Talks given about Gutenberg, including slides and videos as they are available.
+Talks were given about Gutenberg, including slides and videos as they are available.
```
Comment (swissspidy) [inline docs/contributors/outreach.md]: This changes the meaning of this description and is incorrect IMHO

Comment (murshed) [reply docs/contributors/outreach.md]: Thanks :)

File: docs/contributors/release.md [hunk 3]
```diff
@@ -220,7 +220,7 @@ For each Gutenberg plugin release, WordPress trunk should be synchronized with t
 4. Check out all the files from the release branch: `git checkout release/x.x -- .`.
 5. Commit all changes to the `wp/trunk` branch with `git commit -m "Merge changes published in the Gutenberg plugin vX.X release"` and push to the repository.
 6. Update the `CHANGELOG.md` files of the packages with the new publish version calculated and commit to the `wp/trunk` branch.
-Aassuming the package versions are written using this format `major.minor.patch`, make sure to bump at least the `minor` version number. For example, if the CHANGELOG of the package to be released indicates that the next unreleased version is `5.6.1`, choose `5.7.0` as a version in case of `minor` version.
+Assuming the package versions are written using this format `major.minor.patch`, make sure to bump at least the `minor` version number. For example, if the CHANGELOG of the package to be released indicates that the next unreleased version is `5.6.1`, choose `5.7.0` as a version in case of `minor` version.
```
Comment (murshed) [inline docs/contributors/release.md]: @swissspidy,
Please check others when you have time.

Thanks so much :)

File: docs/contributors/git-workflow.md (hunk 1 above)
Comment (gziolo) [inline docs/contributors/git-workflow.md:46]:
```suggestion
You will need to add an `upstream` remote in order to keep your fork updated.
```

File: docs/contributors/outreach.md (hunk 2 above)
Comment (gziolo) [inline docs/contributors/outreach.md:38]:
```suggestion
Talks given about Gutenberg, including slides and videos as they are available.
```

**Review by gziolo — APPROVED** [review]
I reverted changes pointed out by @swissspidy. The rest looks good, thank you 👍

**Review by gziolo — COMMENTED** [review]
Well, there are some merge conflicts. It looks like some updated lines were removed or rephrased.

**Comment by carolinan** [comment]
@murshed Hi Fahim, do you have time to solve the merge conflicts?

File: docs/contributors/git-workflow.md [hunk 4]
```diff
@@ -23,9 +23,9 @@ The workflow is documented in greater detail in the [repository management](/doc
...
+When many different people are working on a project simultaneously, pull requests can go stale quickly. A "stale" pull request is one that is no longer up to date with the main-line of development, and it needs to be updated before it can be merged into the project.
 
-There are two ways to do this: merging and rebasing. In Gutenberg, the recommendation is to rebase. Rebasing means rewriting your changes as if they're happening on top of the main line of development. This ensures the commit history is always clean and linear. Rebasing can be performed as many times as needed while you're working on a pull request. **Do share your work early on** by opening a pull request and keeping your history rebase as you progress.
+There are two ways to do this: merging and rebasing. In Gutenberg, the recommendation is to rebase. Rebasing means rewriting your changes as if they're happening on top of the main-line of development. This ensures the commit history is always clean and linear. Rebasing can be performed as many times as needed while you're working on a pull request. **Do share your work early on** by opening a pull request and keeping your history rebase as you progress.
```
Comment (swissspidy) [inline docs/contributors/git-workflow.md:28]: I don't think this change is correct? It's `main line`, not `main-line`

File: docs/contributors/outreach.md [hunk 5]
```diff
@@ -4,7 +4,7 @@ This includes articles, talks, demos and anything the community is doing to disc
 
 ## Articles
 
-A short list of useful articles around defining, extending, and contributing to Gutenberg.
+A short-list of useful articles around defining, extending, and contributing to Gutenberg.
```
Comment (swissspidy) [inline docs/contributors/outreach.md:7]: Don't think this is correct.

"This is a short list of [..]" is absolutely fine.

So, no hyphen.

File: docs/contributors/release.md [hunk 6]
```diff
@@ -60,7 +60,7 @@ Releasing the first release candidate for this milestone (`x.x`) involves:
 
 ##### Creating the Release Branch
 
-For each milestone (let's assume it's `x.x` here), a release branch is used to release all RCs and minor releases. For the first RC of the milestone, a release branch is created from master.
+For each milestone (let's assume it's `x.x` here), a release branch is used to release all RCs and minor releases. For the first RC of the milestone, a release branch is created from the master.
```
Comment (swissspidy) [inline docs/contributors/release.md:63]: `master` refers to the primary branch (which was now renamed to `trunk`, FWIW), not "the master", so this is incorrect.

Now it should be "a release branch is created from `trunk`"

File: docs/contributors/release.md [hunk 7]
```diff
@@ -106,7 +106,7 @@ If a bug is found in a release candidate and a fix is committed to `master`, we
 7. Merge the version bump pull request and avoid removing the release branch.
 8. Follow the steps in [build the plugin](#build-the-plugin) and [publish the release on GitHub](#publish-the-release-on-github).
 
-You can copy the existing changelog from the previous release candidate. Let other contributors know that a new release candidate has been released in the [`#core-editor` channel](https://wordpress.slack.com/messages/C02QB2JS7) and the call for testing post.
+You can copy the existing changelog from the previous release candidate. Let other contributors know that a new release candidate has been released in the [`#core-editor` channel](https://wordpress.slack.com/messages/C02QB2JS7) and the call for the testing post.
```
Comment (swissspidy) [inline docs/contributors/release.md:109]: It's `a 'call for testing' post`, not `call for the testing post`, so this change is incorrect as it differs in meaning.

File: docs/contributors/release.md [hunk 8]
```diff
@@ -209,7 +209,7 @@ The Gutenberg repository mirrors the [WordPress SVN repository](https://make.wor
 
 ### Synchronizing WordPress Trunk
 
-For each Gutenberg plugin release, WordPress trunk should be synchronized with this release. This involves the following steps that are automated with `./bin/commander npm-stable` command:
+For each Gutenberg plugin release, the WordPress trunk should be synchronized with this release. This involves the following steps that are automated with `./bin/commander npm-stable` command:
```
Comment (swissspidy) [inline docs/contributors/release.md:212]: This refers to the `trunk` folder in SVN, _not_ "the trunk" as in "the trunk of a car", so this has different meaning.

Let's leave this as "WordPress trunk"

**Comment by skorasaurus** [comment]
thanks for your original PR. There's been a bit of changes in the docs since your PR, so I've taken what is still relevant and put them in a new Pull request, https://github.com/WordPress/gutenberg/pull/35268

**Pull request closed without merging**